src/script/interpreter.cpp

```diff
@@ -2061,6 +2062,70 @@ uint256 ComputeTaprootMerkleRoot(Span<const unsigned char> control, const uint25
     return k;
 }
 
+bool VerifyAnnex(const std::vector<unsigned char>& annex_stack, ScriptExecutionData& execdata)
+{
+    CDataStream annex(MakeByteSpan(annex_stack), ANNEX_SER_TYPE, ANNEX_SER_VERSION);
+
+    /* Pop annex tag */
+    uint8_t annex_tag;
+    annex.read(AsWritableBytes(Span{&annex_tag, 1}));
+    assert(annex_tag == ANNEX_TAG);
+
+    /* Annex record are encoded as:
+     * `VarInt` + 1 bytes: type
+     * `VarInt` OR 0 bytes: length
+     * `length`-bytes: value
+     */
+    while (!annex_stack.empty()) {
+
+        /* `VarInt`-bytes to encode the higher-mask type
+         * 1-byte to encode the lower-mask type
+         * 8192 records with 2 bytes.
+         * 1056767 records with 3 bytes.
+         * 40456331200 records with 4 bytes.
+         * See `VarIntMode` comment in serialize.h for more.
+         */
+        uint64_t nRecordHigherMask = 0;
```

Comment:
1. What does "higher-mask type" mean?
2. Why using the ugly variable naming notation? :)

Reply:
would like some text on the masking myself

Reply:
Completely updated the code here, there should be new text. Dropped "higher-mask type" reference. On the ugly variable naming notation, sounds to me the one in vigor for a bunch of `interpreter.cpp`, see `CheckSequence()` though feel free to suggest something else.